### ISSUE TITLE
allow cloning/templating of singletons

### ISSUE THREAD

**Issue by hubret** [agentejo/cockpit#1136]
collections can be used as templates to quickly add fields to new collections. it would be swell if singletons could also copy existing templates of other singletons or even collections.

**Comment by johnfoley** [comment]
Thanks for your mail. I am out of the office until Monday 24 March. For any urgent matters please do not hesitate to contact me on my mobile, otherwise I will respond to your mail on Monday.

**Comment by hubret** [comment]
it seems like regions do the trick

**Comment by pauloamgomes** [comment]
Sounds a good thing to have. In the meantime and as a quick workaround outside the UI, you can duplicate the PHP file in the storage/singleton folder and rename both the file and the singleton name (in the code).

**Comment by stevrons** [comment]
Paulo, so I tried this and I think I am missing maybe a detail. 

So I have an existing Singleton in which I have numerous components. I want to duplicate it so i went to the /storage/singleton folder, duplicated the php file, renamed the duplicated file, and inside the new file I changed the "name" field to be whatever I wanted it to be. 

When I go into cockpit under Singletons, I see the new singleton there, but none of the components from the original singleton are showing.

**Comment by pauloamgomes** [comment]
You need also to change the _id field, e.g:

<img width="474" alt="image" src="https://user-images.githubusercontent.com/102261/83305194-90b98b00-a200-11ea-99d1-5d4339b77663.png">

**Comment by stevrons** [comment]
thanks so much for the reply.

So after I duplicate the singleton file, and then I change file name of the new file, and then change code at name, label, and _id (as per your screenshot) the result is I have a singleton in Cockpit that does not have the same "components" as the original. 

When you say change _id you mean simply change the beginning of it correct?  So change from settings5d945e95a94cf  to    settingsclone5d945e95a94cf  ?

**Comment by pauloamgomes** [comment]
Yes, the same singleton structure should be there, but not the singleton
values...
